.github/workflows/adapter-test.yaml

```diff
@@ -0,0 +1,333 @@
+              exit 1
+            fi
+
+            # Get changed files between base and head
+            echo "Comparing $BASE_SHA..$HEAD_SHA for adapter changes"
+            CHANGED_FILES=$(git diff --name-only "$BASE_SHA" "$HEAD_SHA" | grep "^adapters/.*\.ts$")
+          else
+            echo "Unsupported event type: $EVENT_NAME"
+            exit 1
+          fi
+
+          if [ -z "$CHANGED_FILES" ]; then
+            echo "No adapter files found to test. Skipping tests."
+            exit 0
+          fi
+
+          echo "Found adapter files to test:"
+          echo "$CHANGED_FILES"
+
+          # Store the list of changed files (newline-separated)
+          echo "changed_files<<EOF" >> $GITHUB_OUTPUT
+          echo "$CHANGED_FILES" >> $GITHUB_OUTPUT
+          echo "EOF" >> $GITHUB_OUTPUT
+
+      - name: Setup Deno
+        uses: denoland/setup-deno@v2
+        with:
+          deno-version: v2.x
+
+      - name: Set test address
+        id: parse
+        env:
+          COMMENT_BODY: ${{ github.event.comment.body }}
+        run: |
+          # Default address
+          ADDRESS="0x69155e7ca2e688ccdc247f6c4ddf374b3ae77bd6"
+
+          # If this is a comment event, try to extract address from comment
+          if [ "${{ github.event_name }}" = "issue_comment" ]; then
+            echo "Parsing comment for address..."
+            # Look for Ethereum address pattern in comment (0x followed by 40 hex chars)
+            COMMENT_ADDRESS=$(echo "$COMMENT_BODY" | grep -oE '0x[a-fA-F0-9]{40}' | head -n1)
+            
+            if [ -n "$COMMENT_ADDRESS" ]; then
+              ADDRESS="$COMMENT_ADDRESS"
+              echo "Found address in comment: $ADDRESS"
+            else
+              echo "No valid address found in comment, using default"
+            fi
+          else
+            echo "Using default address for PR event"
+          fi
+
+          # Mask the address to prevent it from appearing in logs
+          echo "::add-mask::$ADDRESS"
+          echo "address=$ADDRESS" >> $GITHUB_OUTPUT
+
+      - name: Cache Deno dependencies
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.cache/deno
+            ~/.deno
+            node_modules
+          key: ${{ runner.os }}-deno-${{ hashFiles('**/deno.lock', '**/deno.json', '**/package.json') }}
+          restore-keys: |
+            ${{ runner.os }}-deno-
+
+      - name: Install dependencies
+        run: deno install
+
+      - name: Run adapter tests
+        id: test
+        env:
+          CHANGED_FILES: ${{ steps.check-changes.outputs.changed_files }}
+          POINTS_ADDRESS: ${{ steps.parse.outputs.address }}
+        run: |
+          if [[ ! "$POINTS_ADDRESS" =~ ^0x[a-fA-F0-9]{40}$ ]]; then
+            echo "Error: Invalid address format"
+            exit 1
+          fi
+
+          # Initialize overall results
+          overall_exit_code=0
+          NO_POINTS_ADDRESS="0x0000000000000000000000000000000000000000"
```

Comment:
hmm there are points apis out there where null address has points (probably for testing) or their api just rejects the address completely, `0x9999999999999999999999999999999999999999` would be better maybe